app/src/test/java/com/example/main_animal_villa/CharacterTests.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.main_animal_villa
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.animal_villa.MainViewModel
+import com.animal_villa.dto.Character
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.impl.annotations.MockK
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertTrue
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.*
+import org.junit.rules.TestRule
+import java.util.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+class CharacterTests {
+
+    @get:Rule
+    var rule : TestRule = InstantTaskExecutorRule()
+    var character : Character? = Character()
+
+    lateinit var characterService : CharacterService
+
+    lateinit var mvm : MainViewModel
+
+    @MockK
+    lateinit var mockCharacterService : CharacterService
+
+    private val mainThreadSurrogate = newSingleThreadContext("Main Thread")
+
+    @Before
+    fun initMocksAndMainThread()    {
+        MockKAnnotations.init(this)
+        Dispatchers.setMain(mainThreadSurrogate)
+    }
+
+    @After
+    fun tearDown()  {
+        Dispatchers.resetMain()
+        mainThreadSurrogate.close()
+    }
+
+    @Test
+    fun `given character data is available when I name my character Billy and give them the janitor occupation then Character class should have Billy as name and Janitor as occupation` () = runTest {
+        givenCharacterServiceIsInitialized()
+        whenCharacterDataAreReadAndParsed()
+        thenTheCharacterShouldHaveTheNameBillyAndTheOccupationJanitor()
+    }
+
+    @Test
+    private fun givenCharacterServiceIsInitialized()    {
+        characterService = CharacterService()
+    }
```

Comment:
CharacterService() does not actually exist yet. TDD is used to help you understand how a program will come together to produce the test that you have given it. Based off your examples, you can make unit tests with your text speed settings before creating the classes to help you understand what functions might need to be made and what variables would need to be accessed to produce the results that are expected.